File: app/src/main/java/com/scatl/uestcbbs/util/ServiceUtil.java
package com.scatl.uestcbbs.util;

import android.app.ActivityManager;
import android.content.Context;

import java.util.ArrayList;

/**
 * author: sca_tl
 * description:
 * date: 2019/8/13 17:18
 */
public class ServiceUtil {

    public static boolean isServiceRunning(Context context, String serviceName) {
        if (serviceName == null) return false;
        ActivityManager activityManager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        ArrayList<ActivityManager.RunningServiceInfo> runningServiceInfos = (ArrayList<ActivityManager.RunningServiceInfo>) activityManager.getRunningServices(20);
        for (int i = 0; i < runningServiceInfos.size(); i ++) {
            if (runningServiceInfos.get(i).service.getClassName().equals(serviceName)) {
                return true;
            }
        }
        return false;
    }
}

File: util/src/main/java/com/scatl/util/ServiceUtil.kt
package com.scatl.util

import android.app.ActivityManager
import android.content.Context

/**
 * created by sca_tl at 2023/3/18 20:24
 */
object ServiceUtil {

    @JvmStatic
    fun isServiceRunning(context: Context, serviceName: String?): Boolean {
        if (serviceName == null) {
            return false
        }
        val activityManager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val runningServiceInfos = activityManager.getRunningServices(20) as ArrayList<ActivityManager.RunningServiceInfo>
        for (i in runningServiceInfos.indices) {
            if (runningServiceInfos[i].service.className == serviceName) {
                return true
            }
        }
        return false
    }

}